File: frontend/src/types/investment.ts
export interface InvestmentData {
  investment_type: 'sip' | 'rd' | 'fd' | 'ppf' | 'nps'
  monthly_amount: number
  annual_return_rate: number
  investment_period_years: number
  age?: number
  inflation_rate?: number
}

export interface YearlyBreakdown {
  year: number
  invested_amount: number
  corpus_value: number
  inflation_adjusted_value: number
  real_returns: number
}

export interface InvestmentResult {
  investment_type: string
  total_invested: number
  final_corpus: number
  total_returns: number
  inflation_adjusted_corpus: number
  real_returns: number
  yearly_breakdown: YearlyBreakdown[]
  ai_explanation: string
  recommendations: string[]
}

export interface InsuranceData {
  age: number
  annual_income: number
  dependents: number
  existing_coverage?: number
  health_conditions?: string[]
}

export interface InsuranceResult {
  recommended_life_cover: number
  recommended_health_cover: number
  estimated_premium: number
  coverage_gap: number
  ai_explanation: string
  recommendations: string[]
}

File: backend/app/models/investment.py
from pydantic import BaseModel, Field
from typing import List, Optional
from enum import Enum

class InvestmentType(str, Enum):
    SIP = "sip"
    RD = "rd"
    FD = "fd"
    PPF = "ppf"
    NPS = "nps"

class InvestmentRequest(BaseModel):
    investment_type: InvestmentType
    monthly_amount: float = Field(..., gt=0, description="Monthly investment amount in INR")
    annual_return_rate: float = Field(..., gt=0, le=50, description="Expected annual return rate in %")
    investment_period_years: int = Field(..., gt=0, le=50, description="Investment period in years")
    age: Optional[int] = Field(None, ge=18, le=100, description="Current age")
    inflation_rate: Optional[float] = Field(6.0, description="Expected inflation rate in %")

class YearlyBreakdown(BaseModel):
    year: int
    invested_amount: float
    corpus_value: float
    inflation_adjusted_value: float
    real_returns: float

class InvestmentResult(BaseModel):
    investment_type: str
    total_invested: float
    final_corpus: float
    total_returns: float
    inflation_adjusted_corpus: float
    real_returns: float
    yearly_breakdown: List[YearlyBreakdown]
    ai_explanation: str
    recommendations: List[str]

class InsuranceRequest(BaseModel):
    age: int = Field(..., ge=18, le=65)
    annual_income: float = Field(..., gt=0)
    dependents: int = Field(..., ge=0)
    existing_coverage: Optional[float] = Field(0)
    health_conditions: Optional[List[str]] = Field([])

class InsuranceResult(BaseModel):
    recommended_life_cover: float
    recommended_health_cover: float
    estimated_premium: float
    coverage_gap: float
    ai_explanation: str
    recommendations: List[str]